src/main/java/blusunrize/immersiveengineering/client/IEManual.java

```diff
@@ -189,20 +186,25 @@ private static Pair<String, List<SpecialElementData>> getMineralVeinTexts()
 		for(RecipeHolder<MineralMix> holder : mineralsToAdd)
 		{
 			final MineralMix mineral = holder.value();
-			String dimensionString;
-			if(mineral.dimensions!=null&&!mineral.dimensions.isEmpty())
-			{
-				StringBuilder validDims = new StringBuilder();
-				for(ResourceKey<DimensionType> dim : mineral.dimensions)
-					validDims.append((!validDims.isEmpty())?", ": "")
-							.append("<dim;")
-							.append(dim.location())
-							.append(">");
-				dimensionString = I18n.get("ie.manual.entry.mineralsDimValid", toName.apply(holder), validDims.toString());
-			}
-			else
-				dimensionString = I18n.get("ie.manual.entry.mineralsDimAny", toName.apply(holder));
+			// Assemble text for biome limitations
+			String biomeText = mineral.biomeTagPredicates.stream().map(
+					biomeTagPredicate -> biomeTagPredicate.tags().stream().map(biomeTagKey -> {
+						// translate biome tags where possible, fall back to location path otherwise
+						String key = "ie.manual.entry.minerals.biome_tag."+biomeTagKey.location();
```

Comment:
I would suggest following [this](https://github.com/TelepathicGrunt/Bumblezone/blob/9dc05697ed23569d941c3f8685f51d41944d7361/common/src/main/resources/assets/the_bumblezone/lang/en_us.json#L1240) format for the keys. While it is not widely used for biome tags AFAICT, the general format will become official-ish with https://github.com/neoforged/NeoForge/pull/135.